support/scripts/uk-libinfo.py

```diff
@@ -0,0 +1,265 @@
+				self._tname = "UKCONFIG"
+				self._tdesc = "Unikraft .config"
+				self._data  = (True, data)
+			case 0x000A:
+				self._tname = "UKCONFIGGZ"
+				self._tdesc = "Unikraft .config.gz"
+				self._data  = (True, data)
+			case 0x000B:
+				self._tname = "COMPILER"
+				self._tdesc = "Compiler (CC)"
+				self._data  = self._data_decode(data, 'ascii')
+			case 0x000C:
+				self._tname = "COMPILEDATE"
+				self._tdesc = "Compile date"
+				self._data  = self._data_decode(data, 'ascii')
+			case 0x000D:
+				self._tname = "COMPILEDBY"
+				self._tdesc = "Compiled by"
+				self._data  = self._data_decode(data, 'ascii')
+			case 0x000E:
+				self._tname = "COMPILEDBYASSOC"
+				self._tdesc = "Compiled by (association)"
+				self._data  = self._data_decode(data, 'ascii')
+			case 0x000F:
+				self._tname = "COMPILEOPTS"
+				self._tdesc = "Compile options"
+				flags = int.from_bytes(data, byteorder)
+				decflags = []
+
+				# Decode UKLI_REC_CO_* flags
+				if self._flag_isset(flags, 0):
+					decflags.append("PIE")
+				if self._flag_isset(flags, 1):
+					decflags.append("DCE")
+				if self._flag_isset(flags, 2):
+					decflags.append("LTO")
+
+				self._data  = (False, ", ".join(decflags))
+			case _:
+				self._tname = int(type)
+				self._tdesc = "Unknown record"
+				self._data  = (True, data)
+
+	def _data_decode(self, data, encoding='ascii'):
+		try:
+			return (False, data.decode(encoding))
+		except:
+			# decoding failed, treat as raw data
+			return (True, data)
+
+	def _flag_isset(self, flags, flagpos):
+		return flags & (0x1 << flagpos)
+
+	def type(self):
+		return self._tnum
+
+	def type_name(self):
+		return self._tname
+
+	def type_description(self):
+		return self._tdesc
+
+	def data(self):
+		return self._data[1]
+
+	def data_is_raw(self):
+		return self._data[0]
+
+	def str_data(self):
+		if self.data_is_raw():
+			str_data = "<binary data>"
+		else:
+			str_data = str(self.data())
+		return str_data
+
+	def __str__(self):
+		return "{:}: {:}".format(self._tdesc, self.str_data())
+
+class UKLibraryInfoHeader:
+	def __init__(self, version, byteorder):
+		self._version = version
+		self._libname = None
+		self._recs    = []
+
+	def add_rec(self, rec):
+		if self._libname == None and rec.type() == 0x0001: # LIBNAME
+			self._libname = rec.data()
+		else:
+			self._recs.append(rec)
+
+	def __str__(self):
+		ret = ""
+		if self._libname == None:
+			for rec in self._recs:
+				str_type = "{:}:".format(rec.type_description())
+				ret += "{:30}{:}\n".format(str_type, rec.str_data())
+		else:
+			ret = "{:}:\n".format(self._libname)
+			for rec in self._recs:
+				str_type = "{:}:".format(rec.type_description())
+				ret += "  {:28}{:}\n".format(str_type, rec.str_data())
+		return ret
+
+	def get_recs(self):
+		return self._recs
+
+	def libname(self):
+		return self._libname
+
+def parse_uklibinfo(data, byteorder=sys.byteorder):
+	if not data:
+		raise Exception("No uklibinfo data found for parsing")
+	hdr_hdr_len = (4 + 2)
```

Comment:
Some explanatory comments, similarly to the ones provided in similar cases below (eg line 181), would be useful here.

Reply:
👍🏻  Sure, I'll add.